packages/opentelemetry-plugin-postgres/src/pg.ts

```diff
@@ -0,0 +1,160 @@
+        'query',
+        this._getClientQueryPatch() as never
+      );
+    }
+    return this._moduleExports;
+  }
+  protected unpatch(): void {
+    if (this._moduleExports.Client.prototype.query) {
+      shimmer.unwrap(this._moduleExports.Client.prototype, 'query');
+    }
+  }
+
+  private _getClientQueryPatch() {
+    const plugin = this;
+    return (original: typeof pgTypes.Client.prototype.query) => {
+      plugin._logger.debug(
+        `Patching ${PostgresPlugin.component}.Client.prototype.query`
+      );
+      return function query(this: pgTypes.Client, ...args: unknown[]) {
+        // setup span
+        let callbackProvided: boolean =
+          args.length > 1 && typeof args[args.length - 1] === 'function';
+        const span = plugin._tracer.startSpan(
+          `${PostgresPlugin.component}.query`,
+          {
+            kind: SpanKind.CLIENT,
+            parent: plugin._tracer.getCurrentSpan() || undefined,
+            attributes: {
+              [AttributeNames.COMPONENT]: PostgresPlugin.component,
+              [AttributeNames.PG_HOST]: (this as any).connectionParameters.host,
+              [AttributeNames.PG_PORT]: (this as any).connectionParameters.port,
+            },
+          }
+        );
+
+        try {
+          if (typeof args[0] === 'string') {
+            span.setAttribute(AttributeNames.PG_TEXT, args[0]);
+            if (args[1] instanceof Array) {
+              span.setAttribute(AttributeNames.PG_VALUES, args[1]);
+              if (callbackProvided) {
+                args[2] = plugin._tracer.bind(args[2]);
+              }
+            } else {
+              if (callbackProvided) {
+                args[1] = plugin._tracer.bind(args[1]);
+              }
+            }
+          } else {
+            const config = args[0] as pgTypes.QueryConfig & {
+              callback?: PostgresCallback;
+            };
+            if (typeof config.name === 'string') {
+              span.setAttribute(AttributeNames.PG_PLAN, config.name);
+            } else {
+              if (typeof config.text === 'string') {
+                span.setAttribute(AttributeNames.PG_TEXT, config.text);
+              }
+              if (config.values instanceof Array) {
+                span.setAttribute(AttributeNames.PG_VALUES, config.values);
+              }
+            }
+
+            if (callbackProvided) {
+              if (typeof args[1] === 'function') {
+                args[1] = plugin._tracer.bind(args[1]);
+              } else if (typeof args[2] === 'function') {
+                args[2] = plugin._tracer.bind(args[2]);
+              }
+            } else if (
+              config.callback &&
+              typeof config.callback === 'function'
+            ) {
+              callbackProvided = true;
+              config.callback = plugin._tracer.bind(config.callback);
+            }
+          }
+        } catch (e) {
+          plugin._logger.warn(
```

Comment:
Does it makes sense to open an issue on https://github.com/open-telemetry/opentelemetry-specification, about what statues to set in case of databases. Looks like most the [status codes](https://github.com/open-telemetry/opentelemetry-specification/blob/597718b3fcfaf10bcf45d93f99b66f94a28048cb/specification/api-tracing.md#statuscanonicalcode) are applicable to `HTTP` and `GRPC` requests.